workers/social/lib/social/models/foreignauth.coffee

```diff
@@ -0,0 +1,131 @@
+jraphical = require 'jraphical'
+KodingError = require '../error'
+{ parseClient } = require './utils'
+
+module.exports = class JForeignAuth extends jraphical.Module
+
+  { secure, ObjectId, signature } = require 'bongo'
+
+  @share()
+
+  @set
+    indexes       :
+      # WARNING! ~ GG
+      # to make this working properly we need a compound index here and since
+      # bongo is not supporting them we need to manually define following:
+      #
+      #   - { username: 1, group: 1, provider: 1, foreignId: 1 } (unique)
+      #
+      provider    : 'sparse'
```

Comment:
if we are going to add a compound index, singular sparse indexes are not required